cmd/registry/cmd/upload/manifest.go

```diff
@@ -63,8 +64,15 @@ func manifestCommand(ctx context.Context) *cobra.Command {
 			if err != nil {
 				log.WithError(err).Fatal("Failed to read manifest")
 			}
-			manifestData, _ := proto.Marshal(manifest)
 
+			// validate the manifest
+			for _, resource := range manifest.GeneratedResources {
+				if err := controller.ValidateResourceEntry(resource); err != nil {
+					log.WithError(err).Fatal("Invalid manifest")
```

Comment:
With this being "Fatal", I think we're only going to report the first error found in a manifest. Let's at least add "TODO: report all errors before returning" so that we can later add more complete error reporting.

Reply:
Thanks, good catch! Made a change instead of deferring it in a TODO.

Now, it will log all the errors found in the manifest and exit with a FATAL log "Manifest contains errors".